skxray/core/fitting/tests/test_xrf_fit.py

```diff
@@ -206,3 +198,40 @@ def test_set_param():
 
     input_param = {'bound_type': 'other', 'max': 13.0, 'min': 9.0, 'value': 11.0}
     _set_parameter_hint('coherent_sct_energy', input_param, compton)
+
+
+def test_pixel_fit_multiprocess():
+    param = get_para()
+    y0 = synthetic_spectrum()
+    x = np.arange(len(y0))
+    pileup_peak = ['Si_Ka1-Si_Ka1', 'Si_Ka1-Ce_La1']
+    elemental_lines = ['Ar_K', 'Fe_K', 'Ce_L', 'Pt_M'] + pileup_peak
+
+    default_area = 1e5
+    elist, matv, area_v = construct_linear_model(x, param, elemental_lines, default_area=default_area)
+    exp_data = np.zeros([2, 1, len(y0)])
+    for i in range(exp_data.shape[0]):
+        exp_data[i,0,:] = y0
+    results = fit_pixel_multiprocess_nnls(exp_data, matv, param,
+                                          use_snip=True)
+    # output area of dict
+    result_map = calculate_area(elist, matv, results,
+                                param, first_peak_area=True)
+
+    # compare input list and output elemental list
+    assert_array_equal(elist, elemental_lines+['compton', 'elastic'])
+
+    # Total len includes all the elemental list, compton, elastic and
+    # two more items, which are summed area of background and r-squared
+    total_len = len(elist) + 2
+    assert_array_equal(results.shape, [2, 1, total_len])
+
+    # same exp data should output same results
+    assert_array_equal(results[0,:,:], results[1,:,:])
+
+    for k, v in six.iteritems(result_map):
+        assert_equal(v[0, 0], v[1, 0])
+        if k in ['snip_bkg', 'r_squared']:
```

Comment:
bkg is not a fitting parameter, and r_squared is just a statistical output. I only compare the fitting parameters, such as area of each peak.